neural_style.py

```diff
@@ -200,7 +201,19 @@ def imread(path):
 
 def imsave(path, img):
     img = np.clip(img, 0, 255).astype(np.uint8)
-    Image.fromarray(img).save(path, quality=95)
+    im_obj = Image.fromarray(img)
+    try:
+        im_obj.save(path, quality=95)
+    except KeyError:
```

Comment:
Actually in the example you gave it would save the file as foo.jgp.png, which would be unlikely to clash with an existing file. I do agree that checking a whitelist extension is probably a better approach.